dask/dataframe/rolling.py

```diff
@@ -118,27 +123,61 @@ def _call_method(self, method_name, *args, **kwargs):
         new_name = 'rolling-' + tokenize(
             self.obj, self._rolling_kwargs(), method_name, args, kwargs)
 
-        # For all but the first chunk, we'll pass the whole previous chunk
-        # in so we can use it to pre-feed our window
-        dsk = {(new_name, 0): (
-            call_pandas_rolling_method_single, (old_name, 0),
-            self._rolling_kwargs(), method_name, args, kwargs)}
-        if self.axis in [0, 'rows']:
-            # roll in the partition direction (will need to access neighbor)
-            tail_name = 'tail-{}-{}'.format(self.window-1, old_name)
-            for i in range(1, self.obj.npartitions + 1):
-                # Get just the needed values from the previous partition
-                dsk[tail_name, i-1] = (tail, (old_name, i-1), self.window-1)
-
+        dsk = {}
+        if self.axis in [1, 'columns'] or self.window <= 1:
+            # This is the easy scenario, we're rolling over columns (or not 
+            # really rolling at all, so each chunk is independent.
+            for i in range(self.obj.npartitions):
                 dsk[new_name, i] = (
-                    call_pandas_rolling_method_with_neighbor,
-                    (tail_name, i-1), (old_name, i),
+                    call_pandas_rolling_method_single, (old_name, i),
                     self._rolling_kwargs(), method_name, args, kwargs)
         else:
-            # no communication needed between partitions for columns
-            for i in range(1, self.obj.npartitions + 1):
+            # This is a bit trickier, we need to feed in information from the
+            # neighbors to roll along rows.
+
+            # Figure out how many we need to look at before and after.
+            if self.center:
+                before = self.window // 2
+                after = self.window - before - 1
+            else:
+                before = self.window - 1
+                after = 0
+
+            head_name = 'head-{}-{}'.format(after, old_name)
+            tail_name = 'tail-{}-{}'.format(before, old_name)
+
+            # First chunk, only look after (if necessary)
+            if after > 0 and self.obj.npartitions > 1:
+                next_partition = (head_name, 1)
+                dsk[next_partition] = (head, (old_name, 1), after)
+            else:
+                #XXX what if there is one partition, is the after wrong?
+                next_partition = self.obj._pd
+            dsk[new_name, 0] = (call_pandas_rolling_method_with_neighbors,
+                self.obj._pd, (old_name, 0), next_partition, 0, after,
+                self._rolling_kwargs(), method_name, args, kwargs)
+
+            # All the middle chunks
+            for i in range(1, self.obj.npartitions-1):
```

Comment:
The pep8 people will eventually get upset with us for not putting spaces around binary operators.  I wouldn't bother to change this (someone will inevitably come around with a pep8 PR) but getting into the practice might smooth things out in the future.